gcp_variant_transforms/libs/generate_report_test.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test cases for generate_report module."""
+
+import csv
+import os
+import unittest
+import vcf
+
+import apache_beam as beam
+from apache_beam.pvalue import AsSingleton
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.transforms import Create
+
+from gcp_variant_transforms.beam_io import vcf_header_io
+from gcp_variant_transforms.libs.generate_report import ConflictsReporter
+from gcp_variant_transforms.transforms import merge_headers
+from gcp_variant_transforms.transforms import merge_header_definitions
+
+
+class GenerateReportTest(unittest.TestCase):
+
+  def _get_vcf_header_from_reader(self, reader, file_name):
+    return vcf_header_io.VcfHeader(infos=reader.infos,
+                                   filters=reader.filters,
+                                   alts=reader.alts,
+                                   formats=reader.formats,
+                                   contigs=reader.contigs,
+                                   file_name=file_name)
+
+  def test_report_no_conflicts(self):
+    lines_1 = [
+        '##INFO=<ID=NS,Number=1,Type=Integer,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample1 Sample2\n'
+    ]
+    lines_2 = [
+        '##FORMAT=<ID=NS,Number=1,Type=Float,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample3\n'
+    ]
+    vcf_reader_1 = vcf.Reader(fsock=iter(lines_1))
+    vcf_reader_2 = vcf.Reader(fsock=iter(lines_2))
+    headers_1 = self._get_vcf_header_from_reader(vcf_reader_1, 'file1')
+    headers_2 = self._get_vcf_header_from_reader(vcf_reader_2, 'file2')
+
+    pipeline = TestPipeline()
```

Comment:
This class should be tested in isolation. Please do not use pipeline for it. It adds dependency to other part of the pipeline which is unnecessary and can be headache to debug (e.g., imagine a scenario where your test fails because some other module used in the pipeline is buggy).

Reply:
Thanks for pointing this out! It is really helpful to test the individual functionality.